Sprint 9 HomeWork.ver1

DmitryBrosalin opened the pull request:
- Реализован API.
- Добавлены новые тесты для обработчиков.

DmitryBrosalin committed:
Sprint 9 HomeWork.ver1
- Реализован API.
- Добавлены новые тесты для обработчиков.

DmitryBrosalin committed:
Sprint 9 HomeWork.ver1
- Реализован API.
- Добавлены новые тесты для обработчиков.

DmitryBrosalin committed:
Sprint 9 HomeWork.ver1
- Реализован API.
- Добавлены новые тесты для обработчиков.

slavrikk commented on src/handlers/BaseHttpHandler.java:
```diff
@@ -0,0 +1,53 @@
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+public class BaseHttpHandler implements HttpHandler {
```
Лучше сделать класс абстрактным, тогда можно будет не переопределять метод `handle`

slavrikk commented on .idea/compiler.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Папку .idea лучше удалить из удаленного репозитория

slavrikk commented on src/handlers/EpicsHandler.java:
```diff
@@ -0,0 +1,136 @@
+import java.util.Map;
+import java.util.Optional;
+
+public class EpicsHandler extends TasksHandler {
```
Думаю будет лучше для всех классов хендлеров наследоваться от `BaseHttpHandler`, двойная вложенность кажется излишней

slavrikk commented on src/handlers/EpicsHandler.java:
```diff
@@ -0,0 +1,136 @@
+                                    exchange.sendResponseHeaders(201, 0);
+                                    exchange.close();
+                                } else {
+                                    String text = ("id введенного эпика не совпадает с id эпика, " +
```
Думаю скобки для строк лишние

slavrikk commented on src/handlers/EpicsHandler.java:
```diff
@@ -0,0 +1,136 @@
+import java.util.Optional;
+
+public class EpicsHandler extends TasksHandler {
+    public EpicsHandler(HttpTaskServer taskServer) {
```
Хендлеры не должны внутри себя хранить сервер, это разные по типу и иерархии сущности

slavrikk commented on src/managers/HttpTaskServer.java:
```diff
@@ -0,0 +1,121 @@
+        taskServer.startServer();
+    }
+
+    static class LocalDateTimeAdapter extends TypeAdapter<LocalDateTime> {
```
Адаптеры лучше вынести в отдельные классы

DmitryBrosalin committed:
Sprint 9 HomeWork.ver2
- Сделал BaseHttpHandler абстрактным
- EpicsHandler и SubtasksHandler теперь наследуются напрямую от BaseHttpHandler
- Убрал лишние скобки для строковых литералов
- Изменил конструкторы Handler-ов, теперь они принимают экземпляр TaskManager
- Вынес адаптеры в отдельные файлы

DmitryBrosalin committed:
Sprint 9 HomeWork.ver3

Delete .idea

DmitryBrosalin committed:
Sprint 9 HomeWork.ver2
- Сделал BaseHttpHandler абстрактным
- EpicsHandler и SubtasksHandler теперь наследуются напрямую от BaseHttpHandler
- Убрал лишние скобки для строковых литералов
- Изменил конструкторы Handler-ов, теперь они принимают экземпляр TaskManager
- Вынес адаптеры в отдельные файлы
- Удалил папку .idea и внес ее в .gitignore
